src/charm_state.py

```diff
@@ -119,10 +124,16 @@ def from_charm(cls, charm: CharmBase) -> "CharmState":
         workers = charm.config.get("webserver_workers")
         threads = charm.config.get("webserver_threads")
         flask_config = {
-            k.removeprefix("flask_"): v for k, v in charm.config.items() if k.startswith("flask_")
+            k.removeprefix("flask_"): v
+            for k, v in charm.config.items()
+            if k.startswith("flask_") and k in KNOWN_CHARM_CONFIG
+        }
+        app_config = {
+            k.removeprefix("flask_"): v
+            for k, v in charm.config.items()
+            if k.startswith("flask_") and k not in KNOWN_CHARM_CONFIG
         }
         try:
-            # typing issue of the pydantic model constructor
```

Comment:
Why did you remove this comment, the "type: ignore" is still needed and there

Reply:
It's accidentally removed during merging, I will restore that, thanks!